app/src/main/java/com/timchenko/playlistmaker/presentation/audioplayer/AudioPlayerViewModel.kt

```diff
@@ -111,29 +112,49 @@ class AudioPlayerViewModel(
 
     fun addTrackInPlaylist(playlist: Playlist, track: Track) {
         if (playlist.tracks.isEmpty()) {
-            addTrackToPlaylist(playlist, track.trackId)
+            addTrackToPlaylist(playlist, track)
             addedToPlaylistState.postValue(PlaylistTrackState.Added(playlist.name))
         }
         else {
             if (playlist.tracks.contains(track.trackId)) {
                 addedToPlaylistState.postValue(PlaylistTrackState.Match(playlist.name))
             }
             else {
-                addTrackToPlaylist(playlist, track.trackId)
+                addTrackToPlaylist(playlist, track)
                 addedToPlaylistState.postValue(PlaylistTrackState.Added(playlist.name))
             }
         }
     }
 
-    private fun addTrackToPlaylist(playlist: Playlist, trackId: Int) {
-        playlist.tracks.add(trackId)
+    private fun addTrackToPlaylist(playlist: Playlist, track: Track) {
+        playlist.tracks.add(track.trackId)
         playlist.tracksCounter += 1
+        playlist.trackTimerMillis += track.trackTimeMillis!!
```

Comment:
🍏 Double bang лучше использовать с осторожностью и только в тех местах, где мы на 200% уверены в том, что не получим NullPointerException.
```suggestion
        playlist.trackTimerMillis += track.trackTimeMillis ?: 0
```